GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/Controllers/PasswordReset/PasswordResetControllerTests.cs

```diff
@@ -0,0 +1,189 @@
+            Assert.AreEqual(1, controllerBuilder.EmailsSent.Count);
+
+            var email = controllerBuilder.EmailsSent.FirstOrDefault();
+            Assert.NotNull(email);
+            Assert.AreEqual(EmailTemplates.SendResetPasswordVerificationEmail, email.TemplateId);
+            Assert.AreEqual(user.EmailAddress, email.EmailAddress);
+        }
+        
+        [Test]
+        [Description("POST: Password reset email is not sent when one has been sent within last 10 minutes")]
+        public void POST_Password_Reset_Email_Is_Not_Sent_When_One_Has_Been_Sent_Within_Last_10_Minutes()
+        {
+            // Arrange
+            User user = new UserBuilder().WithEmailAddress("user@test.com").WithPasswordResetCode("code", VirtualDateTime.Now.AddMinutes(-2)).Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_EmailAddress", "user@test.com");
+
+            var controllerBuilder = new ControllerBuilder<PasswordResetController>();
+            var controller = controllerBuilder
+                .WithLoggedInUser(user)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .WithMockUriHelper()
+                .Build();
+            
+            // Act
+            Assert.Throws<UserRecentlySentPasswordResetEmailWithoutChangingPasswordException>(() => controller.PasswordResetPost(new PasswordResetViewModel()));
+        }
+        
+        [Test]
+        [Description("POST: Password reset email is not sent when there is no user with email address provided")]
+        public void POST_Password_Reset_Email_Is_Not_Sent_When_There_Is_No_User_With_Email_Address_Provided()
+        {
+            // Arrange
+            User user = new UserBuilder().WithEmailAddress("user@test.com").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_EmailAddress", "anotheruser@test.com");
+
+            var controllerBuilder = new ControllerBuilder<PasswordResetController>();
+            var controller = controllerBuilder
+                .WithLoggedInUser(user)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .WithMockUriHelper()
+                .Build();
+            
+            // Act
+            controller.PasswordResetPost(new PasswordResetViewModel());
+            
+            // Assert
+            Assert.IsEmpty(controllerBuilder.EmailsSent);
+        }
+        
+        [Test]
+        [Description("POST: Valid reset code in URL allows user to change their password")]
+        public void POST_Valid_Reset_Code_In_URL_Allows_User_To_Change_Their_Password()
+        {
+            // Arrange
+            User user = new UserBuilder().WithEmailAddress("user@test.com").WithPasswordResetCode("code").Build();

```

Comment:
I don't think we need to specify the email address (in a few tests in this file)